From 262653c976363474233ce74148ed5e4a9b8c894b Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Sat, 10 Feb 2018 15:08:38 +0800
Subject: [PATCH] drm/rockchip: Fix cvbs display err when power off

DPMS status default value should be set according to
whether uboot logo is enabled. If uboot logo is enabled,
DPMS status default value should be set to ON.

Change-Id: I492d76c29687e583771824fcbc5a57455242ec0e
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_tve.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_tve.c b/drivers/gpu/drm/rockchip/rockchip_drm_tve.c
index 23f2ad73b7bc..23eb02585614 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_tve.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_tve.c
@@ -372,8 +372,10 @@ static void check_uboot_logo(struct rockchip_tve *tve)
 
 	vdac = tve_dac_readl(VDAC_VDAC1);
 	/* Whether the dac power has been turned down. */
-	if (vdac & m_DR_PWR_DOWN)
+	if (vdac & m_DR_PWR_DOWN) {
+		tve->connector.dpms = DRM_MODE_DPMS_OFF;
 		return;
+	}
 
 	lumafilter0 = tve_readl(TV_LUMA_FILTER0);
 	lumafilter1 = tve_readl(TV_LUMA_FILTER1);
@@ -385,10 +387,13 @@ static void check_uboot_logo(struct rockchip_tve *tve)
 	 */
 	if (lumafilter0 == tve->lumafilter0 &&
 	    lumafilter1 == tve->lumafilter1 &&
-	    lumafilter2 == tve->lumafilter2)
+	    lumafilter2 == tve->lumafilter2) {
+		tve->connector.dpms = DRM_MODE_DPMS_ON;
 		return;
+	}
 
 	dac_init(tve);
+	tve->connector.dpms = DRM_MODE_DPMS_OFF;
 }
 
 static const struct of_device_id rockchip_tve_dt_ids[] = {
@@ -477,7 +482,6 @@ static int rockchip_tve_bind(struct device *dev, struct device *master,
 	drm_encoder_helper_add(encoder, &rockchip_tve_encoder_helper_funcs);
 
 	connector = &tve->connector;
-	connector->dpms = DRM_MODE_DPMS_OFF;
 	connector->port = dev->of_node;
 	connector->interlace_allowed = 1;
 	ret = drm_connector_init(drm_dev, connector,
-- 
2.35.3

